Scripts/Data/HierarchyData.cs

```diff
@@ -17,13 +17,15 @@ public override int index
 		public int instanceID { private get; set; }
 
 		public HashSet<string> types { get; set; }
+		public bool locked { get; set; }
```

Comment:
Agreed with @mtschoen on this point.  I'd vote to keep the state outside of the model in this case too.